### PR TITLE
Port System.Runtime.CompilerServices new docs

### PR DESCRIPTION
**Pull request by carlossanlop**
@dotnet/area-system-runtime-compilerservices @directhex @gewarren PTAL

### CODE REVIEW

**Review by tannergooding — COMMENTED** [review]
LGTM. @dotnet/interop-contrib should take a peek at the new CallConv

@333fred may want to take a peek at the OverloadResolutionPriority docs as well

**Comment by AaronRobinsonMSFT** [comment]
> @dotnet/interop-contrib should take a peek at the new CallConv

LGTM.

File: xml/System.Runtime.CompilerServices/CallConvSwift.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
   </Base>
   <Interfaces />
   <Docs>
-    <summary>To be added.</summary>
+    <summary>Indicates that a method should using the <see href="https://github.com/apple/swift/blob/main/docs/ABIStabilityManifesto.md#calling-convention">Swift</see>calling convention.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/CallConvSwift.xml:17]:
```suggestion
    <summary>Indicates that a method should using the <see href="https://github.com/apple/swift/blob/main/docs/ABIStabilityManifesto.md#calling-convention">Swift</see> calling convention.</summary>
```

File: xml/System.Runtime.CompilerServices/ParamCollectionAttribute.xml [hunk 2]
```diff
@@ -20,7 +20,7 @@
     </Attribute>
   </Attributes>
   <Docs>
-    <summary>To be added.</summary>
+    <summary>Indicates that a method will allow a variable number of arguments in its invocation.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/ParamCollectionAttribute.xml:23]:
```suggestion
    <summary>Indicates that a method allows a variable number of arguments in its invocation.</summary>
```

File: xml/System.Runtime.CompilerServices/RuntimeFeature.xml [hunk 3]
```diff
@@ -97,7 +97,7 @@
         <ReturnType>System.String</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Represents a runtime feature where byref-like types can be used in Generic parameters.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/RuntimeFeature.xml:100]:
```suggestion
        <summary>Represents a runtime feature where byref-like types can be used in generic parameters.</summary>
```

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 4]
```diff
@@ -147,11 +147,14 @@
         <Parameter Name="type" Type="System.RuntimeTypeHandle" Index="1" FrameworkAlternate="net-9.0" />
       </Parameters>
       <Docs>
-        <param name="target">To be added.</param>
-        <param name="type">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="target">The target data</param>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml:150]:
```suggestion
        <param name="target">The target data.</param>
```

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 5]
```diff
@@ -147,11 +147,14 @@
...
+        <param name="type">The type of box to create.</param>
+        <summary>Create a boxed object of the specified type from the data located at the target reference.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml:152]:
```suggestion
        <summary>Creates a boxed object of the specified type from the data located at the target reference.</summary>
```

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 6]
```diff
@@ -147,11 +147,14 @@
...
+        <returns>A boxed object containing the specified data.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The specified type handle is <see langword="null" />.</exception>
+        <exception cref="T:System.ArgumentException">The specified type cannot have a boxed instance of itself created.</exception>
+        <exception cref="T:System.NotSupportedException">The passed in type is a by-ref-like type.</exception>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml:157]:
```suggestion
        <exception cref="T:System.NotSupportedException">The passed in type is a byref-like type.</exception>
```

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 7]
```diff
@@ -1431,10 +1434,11 @@ This method is intended for compiler use rather than use directly in code. `T` m
         <Parameter Name="type" Type="System.RuntimeTypeHandle" Index="0" FrameworkAlternate="net-9.0" />
       </Parameters>
       <Docs>
-        <param name="type">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="type">The type to get the size of.</param>
+        <summary>Get the size of an object of the given type.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml:1438]:
```suggestion
        <summary>Gets the size of an object of the given type.</summary>
```

**Review by gewarren — APPROVED** [review]

**Comment by 333fred** [comment]
> @333fred may want to take a peek at the OverloadResolutionPriority docs as well

LGTM.

**Review by jkoritzinsky — APPROVED** [review]

**Pull request merged**